api/src/main/java/org/apache/iceberg/Files.java

```diff
@@ -59,9 +59,9 @@ public PositionOutputStream create() {
       }
 
       if (!file.getParentFile().isDirectory() && !file.getParentFile().mkdirs()) {
-        throw new RuntimeIOException(
+        throw new UncheckedIOException(new IOException(String.format(
```

Comment:
Rather than creating an `IOException`, I think this should throw a different exception.